File: Easy/Maximum Sum Problem/maximum-sum-problem.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main(String args[])
        {
            Scanner sc = new Scanner(System.in);
            int t = sc.nextInt();
            while(t-->0)
                {
                    int n = sc.nextInt();
                    Solution ob = new Solution();
                    System.out.println(ob.maxSum(n));
                }
        }
}    
// } Driver Code Ends


//User function Template for Java

class Solution
{
    Map<Integer,Integer> map =new HashMap<>();
    public int moon(int n){
        if(n==0)return 0;
        if(map.containsKey(n))return map.get(n);
        int a= Math.max(n,moon(n/2)+moon(n/3)+moon(n/4));
        map.put(n,a);
        return a;
    }
    public int maxSum(int n) 
    { 
        return moon(n);
    } 
}

File: Easy/Maximum Sum Problem/maximum-sum-problem.py
#User function Template for python3

class Solution:
    def maxSum(self, n): 
        # code here
        def help(n,d):
            if n <= 11:
                d[n] = n
                return d[n]
            if n in d:
                return d[n]
            else:
                first = (n // 2)
                second = (n // 3)
                third = (n // 4)
                if first > 11:
                    d[first] = help(first, d)
                    first = d[first]
                if second > 11:
                    d[second] = help(second, d)
                    second = d[second]
                if third > 11:
                    d[third] = help(third, d)
                    third = d[third]
                d[n] = first + second + third
                return d[n]
        return help(n,{})



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int(input())
    for _ in range(t):
        n = int(input())
        ob = Solution()
        print(ob.maxSum(n))
# } Driver Code Ends